sdk/messaging/azeventgrid/models.go

```diff
@@ -0,0 +1,200 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is regenerated.
+// DO NOT EDIT.
+
+package azeventgrid
+
+import "time"
+
+// AcknowledgeOptions - Array of lock token strings for the corresponding received Cloud Events to be acknowledged.
+type AcknowledgeOptions struct {
+	// REQUIRED; String array of lock tokens.
+	LockTokens []*string
+}
+
+// AcknowledgeResult - The result of the Acknowledge operation.
+type AcknowledgeResult struct {
+	// REQUIRED; Array of LockToken values for failed cloud events. Each LockToken includes the lock token value along with the
+	// related error information (namely, the error code and description).
+	FailedLockTokens []*FailedLockToken
+
+	// REQUIRED; Array of lock tokens values for the successfully acknowledged cloud events.
+	SucceededLockTokens []*string
+}
+
+// AzureCoreFoundationsError - The error object.
+type AzureCoreFoundationsError struct {
```

Comment:
I suspect we might want to give this error type the same treatment that we did in [`azadmin`](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/security/keyvault/azadmin/backup/custom_client.go#L54-L60).